core/app/models/spree/shipment.rb

```diff
@@ -214,23 +214,29 @@ def refresh_rates(shipping_method_filter = ShippingMethod::DISPLAY_ON_FRONT_END)
       return [] unless can_get_rates?
 
       # StockEstimator.new assigment below will replace the current shipping_method
-      original_shipping_method_id = shipping_method.try(:id)
+      original_shipping_method_id = shipping_method.try!(:id)
 
-      self.shipping_rates = Stock::Estimator.new(order).
+      new_rates = Stock::Estimator.new(order).
       shipping_rates(to_package, shipping_method_filter)
```

Comment:
@wuboy0307 
Hmm, the
```
shipping_rates(to_package, shipping_method_filter)
```
used to had new_rates as argument. And now it  uses old ones. This method only sorts rates and sets default one, but wouldn't this be an issue in some cases?

Otherwise, it seems allright. Any thoughts @damianlegawiec, @jitendra, @priyank-gupta